core/storage/base_model/gae_models.py

```diff
@@ -241,12 +242,67 @@ class BaseCommitLogEntryModel(BaseModel):
     version = ndb.IntegerProperty()
 
     @classmethod
-    def _get_instance_id(cls, instance_id, version):
+    def create(
+            cls, entity_id, version, committer_id, committer_username,
+            commit_type, commit_message, commit_cmds, status=None,
+            community_owned=None):
+        """This method returns an instance of the CommitLogEntryModel for a
+        construct with the common fields filled.
+
+        Args:
+            committer_id: str. The user_id of the user who committed the
+                change.
+            committer_username: str. The username of the user who committed the
+                change.
+            commit_type: str. The type of commit. Possible values are in
+                core.storage.base_models.COMMIT_TYPE_CHOICES.
+            commit_message: str. The commit description message.
+            commit_cmds: list(dict). A list of commands, describing changes
+                made in this model, which should give sufficient information to
+                reconstruct the commit. Each dict always contains:
+                    cmd: str. Unique command.
+                and then additional arguments for that command.
+            status: str. The status of the entity after the commit.
+            community_owned: bool. Whether the entity is community_owned after
+                the commit.
+
+        Returns:
+            CommitLogEntryModel. Returns the respective CommitLogEntryModel
+                instance of the construct from which this is called.
+        """
+        if status:
+            return cls(
+                id=cls._get_instance_id(entity_id, version),
+                user_id=committer_id,
+                username=committer_username,
+                commit_type=commit_type,
+                commit_message=commit_message,
+                commit_cmds=commit_cmds,
+                version=version,
+                post_commit_status=status,
+                post_commit_community_owned=community_owned,
+                post_commit_is_private=(
+                    status == feconf.ACTIVITY_STATUS_PRIVATE)
+            )
+        else:
+            return cls(
```

Comment:
Keys should be stable -- don't branch. Just use sensible default values for the other items (public, False, False) if you don't have a rights management system yet.

An alternative is to do what you did for collection_id and set the fields that are optional for each entity in the subclass; this list should be identical to the fields that are defined only in the subclass. But it seems to me like status/community-owned/etc. are generic things and it's probably worth moving them to the base class.

Reply:
Done, made them (public, False, False) for story.